### PR TITLE
cli: Fix ttn-lw-cli failing when not in a TTY

### LINKED ISSUE #1841: Create gateway fails unless run from a TTY
## Summary
When automating the gateway creation process the 'ttn-lw-cli gateway create' command silently fails if not run from a TTY. 

Note that this appears to be the case for gateway creation. Other cli commands (gateways list etc) seems to work correctly when run without a tty.

#### Steps to Reproduce
1. Run the following collection of scripts to simulate running the command without an attached TTY

```
#!/bin/bash
true | ttn-lw-cli gateways create test-gw-$RANDOM --user-id admin
```

#### What do you see now?
The ttn-lw-cli thinks input is coming via a pipe (as there is no stdin attached) and then the inputDecoder throws an error on EOF. It then exits with a return code of 0 (success) even though the command failed to take action on the variables passed in by argument

#### What do you want to see instead?
Gateway should be created successfully

#### Environment
The Things Network Command-line Interface: ttn-lw-cli
Version:             3.4.2
Build date:          2020-01-08T16:30:08Z
Git commit:          2714422a2
Go version:          go1.13.5
OS/Arch:             linux/amd64

#### How do you propose to implement this?
Gratefully catch stdin being disconnected and not expect data for the inputDecoder

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

Closes #1841

#### Changes

- Accept inputDecoder failing with EOF. Return EOF unless any other errors occur

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [X] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [X] Documentation: Relevant documentation is added or updated.
- [X] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: cmd/ttn-lw-cli/commands/gateways.go [hunk 1]
```diff
@@ -259,7 +262,12 @@ var (
 				return err
 			}
 
-			return io.Write(os.Stdout, config.OutputFormat, res)
+			if e := io.Write(os.Stdout, config.OutputFormat, res); e != nil {
```
Comment (rvolosatovs) [inline cmd/ttn-lw-cli/commands/gateways.go]:
```suggestion
			if err := io.Write(os.Stdout, config.OutputFormat, res); err != nil {
```

File: cmd/ttn-lw-cli/internal/io/io.go [hunk 2]
```diff
@@ -124,9 +125,23 @@ func NewJSONDecoder(r io.Reader) Decoder {
...
+		return true
+	}
+	return false
+}
+
 func (r *jsonDecoder) Decode(data interface{}) (paths []string, err error) {
 	t, err := r.rd.ReadByte()
-	if err != nil {
+	if err == io.EOF {
+		return []string{}, nil
```
[the remark this answers is not in the document]
Comment (neoaggelos) [reply cmd/ttn-lw-cli/internal/io/io.go]: Perhaps it is just better to have `inputDecoder` be nil when the underlying reader (`io.Stdin` here) returns EOF, so we would not need the IsOK() check afterwards

File: CHANGELOG.md [hunk 3]
```diff
@@ -23,6 +23,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Updating `supports_class_c` field in the Device General Settings Page in the Console.
 - Updating MQTT pubsub configuration in the Console
 - Handling multiple consequent updates of MQTT pubsub/webhook integrations in the Console.
+- `ttn-lw-cli` failing when input is not a TTY.
```
Comment (htdvisser) [inline CHANGELOG.md]: I would reword this to "CLI trying to read input while none available"
